### ISSUE TITLE
Questions about samplers in mmdetection

### ISSUE THREAD

**Issue by goolhanrry** [open-mmlab/mmdetection#1777]
Thank you for your attention. Here are some questions about the samplers in mmdetection.

1. I notice that `DistributedSampler` in `mmdet.datasets.loader.sampler` is almost the same with `torch.utils.data.DistributedSampler`, why it is necessary to re-implement it?

2. What are the differences between `GroupSampler` and normal `RandomSampler` or `SequentialSampler`? It seems that `GroupSampler` works with `flag` which indicates the aspect ratio of images in datasets but I can't understand how and why it works.

**Comment by hellock** [comment]
1. `shuffle` is not an optional argument in `torch.utils.data.DistributedSampler` before PyTorch 1.2, so we wrote our own DistributedSampler (when the version of PyTorch was only 1.0).
2. `flag` can be used for grouping images in a batch by aspect ratios, which can save memory for training.

**Comment by goolhanrry** [comment]
Thank you !
